src/index.ts

```diff
@@ -35,22 +35,23 @@ function keyDownHandler(event: KeyboardEvent) {
 
   currentTriePosition = newTriePosition
   if (newTriePosition instanceof Leaf) {
+    const target = event.target as HTMLElement
+    const scopedElements = newTriePosition.children.filter(element => element.hasAttribute('data-hotkey-scope'))
+    const nonScopedElements = newTriePosition.children.filter(element => !element.hasAttribute('data-hotkey-scope'))
     let shouldFire = true
-    const elementToFire = newTriePosition.children[newTriePosition.children.length - 1]
-    const hotkeyScope = elementToFire.getAttribute('data-hotkey-scope')
-    if (isFormField(event.target)) {
-      const target = event.target as HTMLElement
-      if (target.id !== elementToFire.getAttribute('data-hotkey-scope')) {
-        shouldFire = false
-      }
-    } else if (hotkeyScope) {
-      shouldFire = false
+    let elementToFire
+    if (isFormField(target)) {
+      elementToFire = scopedElements.find(element => element.getAttribute('data-hotkey-scope') === target.id)
+      shouldFire = elementToFire ? true : false
+    } else {
+      elementToFire = nonScopedElements[nonScopedElements.length - 1]
```

Comment:
We can minimise on the amount of iterations we do here by doing a reverse iteration on the array and checking conditions during that:
 
```suggestion
    let elementToFire
    const formField = isFormField(target)
    for (let i = newTriePosition.children.length - 1; i >= 0; i -= 1) {
      elementToFire = newTriePosition.children[i]
      const scope = elementToFire.getAttribute('data-hotkey-scope')
      if (!formField && !scope || (formField && target.id === scope)) {
        shouldFire = true
        console.log('firing on ', elementToFire)
        break
      }
    }

```

This will iterate from the end of the array, ending early whenever it matches either a non-form-field non-scope or a formfield with a matching scope.

Reply:
@keithamus I think we need a bit of the extra logic to distinguish:
- scoped with valid id
- scoped with bad id (do nothing)
- non-scoped

This change will not pass [this test]. 

[this test]: https://github.com/github/hotkey/pull/51/files#diff-a561630bb56b82342bc66697aee2ad96efddcbc9d150665abd6fb7ecb7c0ab2fL110